### ISSUE TITLE
`functional/prefer-readonly-type ` reports false positive for `Readonly<>` generics

### ISSUE THREAD

**Issue by maxxxxxdlp** [jonaskello/eslint-plugin-functional#386]
# Bug Report

ESLint reports `functional/prefer-readonly-type` rule on for the `a: number;` line in the following snippet:

```ts
type R<T> = Readonly<T>;
type SomeType = R<{
  a: number;
}>;
```

## Expected behavior

The rule should not be reported as type is a non-readonly type is never created.

## Actual behavior

The rules is falsely reported

## Proposed changes

I understand that it might expensive/complex to detect a case like this.

If that is indeed the case, would it be possible to have a regex option like [`ignorePattern`](https://github.com/jonaskello/eslint-plugin-functional/blob/master/docs/rules/options/ignore-pattern.md) to ignore generic arguments when the name of the generic type matches a certain regex?

**Comment by RebeccaStevens** [comment]
We currently don't have support for `Readonly` (see #51).

I've been working on the next release which will add support for this.
